iamctl/pkg/applications/import.go

```diff
@@ -103,39 +106,68 @@ func importApp(importFilePath string, isUpdate bool) error {
 	modifiedFileData := utils.ReplaceKeywords(string(fileBytes), appKeywordMapping)
 
 	if isUpdate {
-		log.Println("Updating application: " + fileInfo.ResourceName)
-		err = utils.SendUpdateRequest("", importFilePath, modifiedFileData, utils.APPLICATIONS)
-	} else {
-		log.Println("Creating new application: " + fileInfo.ResourceName)
-		err = utils.SendImportRequest(importFilePath, modifiedFileData, utils.APPLICATIONS)
+		return updateApplication(importFilePath, modifiedFileData, fileInfo)
 	}
+	return importApplication(importFilePath, modifiedFileData, fileInfo)
+}
+
+func updateApplication(importFilePath string, modifiedFileData string, fileInfo utils.FileInfo) error {
+
+	log.Println("Updating application: " + fileInfo.ResourceName)
+	err := utils.SendUpdateRequest("", importFilePath, modifiedFileData, utils.APPLICATIONS)
 	if err != nil {
-		return fmt.Errorf("error when importing application: %s", err)
+		utils.UpdateFailureSummary(utils.APPLICATIONS, fileInfo.ResourceName)
+		return fmt.Errorf("Error when updating application: %s", err)
 	}
+	utils.UpdateSuccessSummary(utils.APPLICATIONS, utils.UPDATE)
+	log.Println("Application updated successfully.")
+	return nil
+}
+
+func importApplication(importFilePath string, modifiedFileData string, fileInfo utils.FileInfo) error {
+
+	log.Println("Creating new application: " + fileInfo.ResourceName)
+	err := utils.SendImportRequest(importFilePath, modifiedFileData, utils.APPLICATIONS)
+	if err != nil {
+		utils.UpdateFailureSummary(utils.APPLICATIONS, fileInfo.ResourceName)
+		return fmt.Errorf("Error when importing application: %s", err)
+	}
+
+	if authenticated, err := isAuthenticationApp(modifiedFileData); err != nil {
+		fmt.Println("error occurred:", err)
+	} else if authenticated {
+		utils.IndicateNewSecretGenerated(fileInfo.ResourceName)
+	}
+	utils.UpdateSuccessSummary(utils.APPLICATIONS, utils.IMPORT)
 	log.Println("Application imported successfully.")
 	return nil
 }
 
-func removeDeletedDeployedApps(localFiles []os.FileInfo) {
+func removeDeletedDeployedApps(localFiles []os.FileInfo, importFilePath string) {
 
 	// Remove deployed applications that do not exist locally.
 	deployedApps := getAppList()
 deployedResources:
 	for _, app := range deployedApps {
 		for _, file := range localFiles {
-			if app.Name == utils.GetFileInfo(file.Name()).ResourceName {
+			isToolManagementApp, err := isToolMgtApp(file, importFilePath)
+			if err != nil {
```

Comment:
In case of an error, better to skip deleting the app since we do not know for sure whether it is the management app or not.